src/functTest/groovy/com/bmuschko/gradle/docker/tasks/DockerMultipleClientFunctionalTest.groovy

```diff
@@ -0,0 +1,53 @@
+package com.bmuschko.gradle.docker.tasks
+
+import com.bmuschko.gradle.docker.AbstractGroovyDslFunctionalTest
+import org.gradle.testkit.runner.BuildResult
+
+class DockerMultipleClientFunctionalTest extends AbstractGroovyDslFunctionalTest {
+    def "DockerClient respects task configuration"() {
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.DockerClient
+
+            task dockerClient1(type: DockerClient) {
+                onNext { client ->
+                    if (client != null) {
+                        logger.quiet "config: " + client.dockerClientConfig.toString()
+                    } else {
```

Comment:
I think we can just remove the `else` blocks and assert the "config" message. The "config" message already verifies that the client is not null.